src/core/Akka/Configuration/Config.cs

```diff
@@ -456,16 +450,20 @@ public virtual Config WithFallback(Config fallback)
             if (fallback == this)
                 throw new ArgumentException("Config can not have itself as fallback", nameof(fallback));
 
-            Config clone = Copy();
-
-            Config current = clone;
+            var mergedRoot = fallback == null
```

Comment:
This code mainly leverages _MergeImmutable_ which follows _Merge_ in its structure with nuance of producing new _HoconValue_ every time it found same key in rhs _HoconObject_. Also there is a little boilerplate with injecting result in final _Config_ instance. I'm keep preserving chain of fallbacks since it is property with public getter. Feel free to ask for more precise explanation if something left unclear.
Also I've missed obvious optimisation as to shortcut entire routine if _fallback_ argument is null. Will fix it now.